app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java

```diff
@@ -16,25 +16,17 @@ public class UpdateSettingsFragment extends BasePreferenceFragment {
                 .apply();
 
         if (checkForUpdates) {
-            checkNewVersionNow();
+            NewVersionWorker.enqueueNewVersionCheckingWork(requireContext(), true);
```

Comment:
I would make this `false` intead of `true`, since the user did not directly ask for an update check, he just enabled them. So if the last check was done a while ago (or never), then it's ok to check again, but if it has been done minutes earlier since the user just disabled and enabled twice in a row, it should not be done. Also, it's better to consider it automatic since we don't want a toast to be shown to the user in this case, as he does not expect feedback I think.

Reply:
I'd keep this `true` because the user enabled the check manually, in other words wants NewPipe to check for new versions. Reporting back, that we did not find any new version *this time* is ok. It shows that the app does something as requested.